### PR TITLE
update config for conceptual versioning

### PR DESCRIPTION
**Pull request by sudeepku**
## Summary

Describe your changes here.

Fixes #Issue_Number (if available)

### CODE REVIEW

**Review by mairaw — CHANGES_REQUESTED** [review]
@sudeepku something went wrong with this config. links between the folders no longer work and path doesn't include core any longer. For example, url looks like https://review.docs.microsoft.com/en-us/dotnet/additional-tools/index, when it should be https://review.docs.microsoft.com/en-us/dotnet/core/additional-tools/index. Also, how do we define the default moniker? We don't want to be 1.0 as it seems to be right now.

File: docfx.json [hunk 1]
```diff
@@ -46,9 +46,11 @@
         ]
       },
       {
+        "group": "netcore",
+        "src": "docs/core",
```
Comment (fenxu) [inline docfx.json]: >"src": "docs/core", [](start = 8, length = 19)

Remove this

File: docfx.json [hunk 2]
```diff
@@ -46,9 +46,11 @@
...
         "files": [
-          "core/**/*.md",
-          "core/**/*.yml"
+          "**/*.md",
+          "**/*.yml"
```
Comment (fenxu) [inline docfx.json]: And revert his

**Review by mairaw — APPROVED** [review]
Merging so we can test the changes on the local branch instead and queue full builds given these are config changes.

**Pull request merged**